src/models/blockchain/block/mod.rs

```diff
@@ -2017,260 +1829,745 @@ pub(crate) mod tests {
             > Thorkil Schmidiger -- will never ask for money:
             I'm not sure it's possible to hit the `BlockValidationError::MutatorSetUpdatePossible` error.
             */
-            // let mut b_now_bad_update = b_now.clone();
-            // let mut tx_kernel_bad_update = TransactionKernelProxy::from(b_now_bad_update.kernel.body.transaction_kernel.clone());
-            // tx_kernel_bad_update.inputs = Vec::new();
-            // tx_kernel_bad_update.outputs = [record_addition_an; 2].to_vec();
-            // b_now_bad_update.kernel.body.transaction_kernel = tx_kernel_bad_update.into_kernel();
-            // // MutatorSetUpdate{additions: [record_addition_an; 2].to_vec(), removals: Vec::new()}.apply_to_accumulator(&mut b_now_bad_update.kernel.body.mutator_set_accumulator);
-            // b_now_bad_update.kernel.appendix = BlockAppendix::new(BlockAppendix::consensus_claims(b_now_bad_update.body()));
-            // // dbg!(&b_now_bad_update.kernel.body.transaction_kernel.inputs);
-            // assert_eq![
-            //     // BlockValidationError::MutatorSetUpdateImpossible,
-            //     BlockValidationError::MutatorSetUpdateIntegrity,
-            //     b_now_bad_update.validate(&b_previous, now, Network::Main).await.err().unwrap()
-            // ];
-
-            let mut b_now_kernel_ts = b_now.clone();
-            let mut tx_kernel_ts = TransactionKernelProxy::from(
-                b_now_kernel_ts.kernel.body.transaction_kernel.clone(),
+            err_variants.retain(|v| v != &BlockValidationError::MutatorSetUpdateImpossible);
+            proptest::proptest!(
+                deterministic,
+                |((b_prev, ts, rness) in n_strategy())|
+                // 60..BFieldElement::MAX
+                {
+                    // prop_assume![num_bigint::BigUint::ZERO != b_prev.header().difficulty.into()]; // super::super::difficulty_control::Difficulty([0; _])
+                    // prop_assume![ts >= (Timestamp::minutes(1) + b_prev.kernel.header.timestamp)];
+                    // prop_assume![!b_prev.body().transaction_kernel.fee.is_negative()];
+
+                    // dbg!(b_prev.kernel.header.timestamp.0.value(), ts);
+                    // // let ts_ =  + b_prev.kernel.header.timestamp;
+                    // dbg!(ts.0.value());
+
+                    // fake_block_successor_with_merged_tx
+                    let t = crate::tests::tokio_runtime().block_on(async {
+                        fake_valid_successor_for_tests(
+                            &b_prev,
+                            ts,
+                            rness,
+                            Network::Main,
+                        ).await.validate(&b_prev, ts, Network::Main).await.is_ok()
+                    });
+                    prop_assert![t];
+                }
             );
```

Comment:
I'm having trouble parsing this. You're defining a proptest within a test, which verifies that arbitrary blocks can be validated successfully? If correct, why? If not, what am I missing?